app/routes/datasets/queries.py

```diff
@@ -95,7 +95,9 @@ async def query_dataset(
     else:
         geometry = None
 
-    data = await _query_dataset(dataset, version, sql, geometry)
+    data: List[Dict[str, Any]] = await _query_dataset(dataset, version, sql, geometry)
+
+    response.headers["Cache-Control"] = "max-age=7200"  # 2h
```

Comment:
We can play around with what cache age works best. Since datasets are immutable we could even go higher.
Only exception would be for raster queries, since the contextual layers might change.

Reply:
Would that work with the "latest" redirect? Would it go back through the cloudfront distribution? For now, since we're using append a lot, we'll probably still have to invalidate at least once a day. I went with two hours since a user could request shortly before a daily update, and we'd want to make sure they can see the new data within a few hours.

Reply:
Good point about the append/ mutable version. Forgot about that.

To answer the other question: the `latest` endpoint always returns a `307` redirect to the correctly versioned endpoint. The client is forced to make a second requested to the current version and the query results will be cached for the actual version number, not for `latest` .